test/functional/mempool_accept.py

```diff
@@ -183,12 +179,12 @@ def run_test(self):
             rawtxs=[raw_tx_1],
         )
 
-        self.log.info('Create a signed "reference" tx for later use')
-        raw_tx_reference = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{'txid': txid_spend_both, 'vout': 0}],
-            outputs=[{node.getnewaddress(): 0.05}],
-        ))['hex']
-        tx = tx_from_hex(raw_tx_reference)
+        self.log.info('Create a "reference" tx for later use')
+        self.wallet.scan_tx(node.decoderawtransaction(raw_tx_spend_both))
+        utxo_to_spend = self.wallet.get_utxo(txid=txid_spend_both)
+        tx = self.wallet.create_self_transfer(utxo_to_spend=utxo_to_spend, sequence=SEQUENCE_FINAL)['tx']
```

Comment:
Why set FINAL when it wasn't set previously?

Reply:
Per default `createrawtransaction` sets the input's nSequence fields to 0xffffffff i.e. it actually was already final before. This is needed because of the following line below:

https://github.com/bitcoin/bitcoin/blob/318c79e8096ce477561ef099835954b95a687685/test/functional/mempool_accept.py#L326

Probably it would make sense in a follow-up to also create CTxIn instances with nSequence=SEQUENCE_FINAL by default in the test framework, to match the behaviour in the bitcoind codebase?

https://github.com/bitcoin/bitcoin/blob/318c79e8096ce477561ef099835954b95a687685/src/primitives/transaction.h#L105-L106

Reply:
ok, fair enough